### DISCUSSION TITLE
install bun on windows without WSL

### DISCUSSION THREAD

**Discussion by elycheikhsmail** [oven-sh/bun#6133] [Ideas]
can I understand why not possible to install bun on Windows without require WSL installed ? like we do with nodejs and denojs !!!

**Comment by melroy89** [comment]
Nah. You need to use wsl. Or go to Linux.

**Reply by abhishekpatelmc** [reply]
Could you please explain why we cannot run Bun on native Windows currently?

**Reply by melroy89** [reply]
Sure, currently the Bun project has code written in Zig that might be very platform specific for Unix systems. Writing a NodeJS alternative like Bun isn't easy, and therefor they now write code that does work on both Linux and Mac systems. Using platform specific calls in order to have the performance you expect with Bun. However, the Bun dev team is busy trying to port all the features and internal working to Windows as well.. See also: https://github.com/oven-sh/bun/blob/main/src/windows.zig

But Bun isn't ready for Windows yet. Sure, you can try to build this project under Windows and see what happens. But again, Bun officially doesn't deliver binaries for Windows at this moment in time.

**Reply by melroy89** [reply]
To understand this situation better, here I just share a small snippet of code remarks by Jarred for Windows (all those 'TODOs' need the be fixed first):

![image](https://github.com/oven-sh/bun/assets/628926/bc49dae3-31ed-4831-8cdb-11922925d685)

Again, this is just a small screenshot, in reality a lot of code lines are needed to change in order to have good Windows native support.  
And the dev team does not have the highest priority on supporting Windows native (which I understand and admire). So it could take months more before Bun will run native under Windows. And even then, it might not perform as good as under Linux.

**Reply by abhishekpatelmc** [reply]
Now I understand, I can't wait to run it on Windows. Until then I will run it on the docker if possible.
Thank you for the explanation @melroy89

**Comment by elycheikhsmail** [comment]
inespected behaivur

**Comment by elycheikhsmail** [comment]
thank you @melroy89 , I understand bun js havr a lot of job to, may running bun on Windows native not prioritt now, but we will always think to bunjs as Node replacement (in the future ) this will including the native installation on Windows.
